qcengine/programs/entos.py

```diff
@@ -0,0 +1,238 @@
+from ..util import execute, popen
+from qcelemental.util import which, safe_version, parse_version, which_import
+
+from .model import ProgramHarness
+
+
+class EntosHarness(ProgramHarness):
+    _defaults = {
+        "name": "entos",
+        "scratch": True,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    def __init__(self, **kwargs):
+        super().__init__(**{**self._defaults, **kwargs})
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which('entos', return_bool=True, raise_error=raise_error, raise_msg='Please install via XXX')
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        which_prog = which('entos')
+        if which_prog not in self.version_cache:
+            with popen([which_prog, '--version']) as exc:
+                exc["proc"].wait(timeout=5)
+            self.version_cache[which_prog] = safe_version(exc["stdout"].split()[2])
+
+        return self.version_cache[which_prog]
+
+    def compute(self, input_data: 'ResultInput', config: 'JobConfig') -> 'Result':
+        """
+        Run entos
+        """
+        # Check if entos executable is found
+        self.found(raise_error=True)
+
+        # Check entos version
+        if parse_version(self.get_version()) < parse_version("0.5"):
+            raise TypeError("entos version '{}' not supported".format(self.get_version()))
+
+        # Setup the job
+        job_inputs = self.build_input(input_data, config)
+
+        # Run entos
+        proc = self.execute(job_inputs)
+
+        if isinstance(proc, FailedOperation):
+            return proc
+        else:
+            # If execution succeeded, collect results
+            result = self.parse_output(proc["outfiles"], input_data)
+            return result
+
+    def execute(self, inputs, extra_outfiles=None, extra_commands=None, scratch_name=None, timeout=None):
+
+        if extra_outfiles is not None:
+            outfiles = ["dispatch.out"].extend(extra_outfiles)
+        else:
+            outfiles = ["dispatch.out"]
+
+        if extra_commands is not None:
+            commands = extra_commands
+        else:
+            commands = inputs["commands"]
+
+        exe_success, proc = execute(commands,
+                                    infiles=inputs["infiles"],
+                                    outfiles=outfiles,
+                                    scratch_location=inputs["scratch_location"],
+                                    scratch_name=scratch_name,
+                                    timeout=timeout
+                                    )
+        proc["outfiles"]["dispatch.out"] = proc["stdout"]
+
+        # Determine whether the calculation succeeded
+        output_data = {}
+        if not exe_success:
```

Comment:
You can use the new `raise UnknownError` commands rather than forming a full `FailedOperation` see Psi4/rdkit.